### ISSUE TITLE
Excel replace cell value cause format lost

### ISSUE THREAD

**Issue by Edward-Zhou** [OfficeDev/office-js#4726]
I have a cell which contains format like below:
![image](https://github.com/user-attachments/assets/3965fce2-7989-4f41-9ba7-b00aba6f2f39)
Now, I want to replace the "Test" with new line in cell. I tried code below:
```  await Excel.run(async (context) => {
    const range = context.workbook.getSelectedRange();
    range.load("values,rowCount,columnCount");
    await context.sync();
    const rowCount = range.rowCount;
    const colCount = range.columnCount;
    range.values = [[range.values[0][0].replace(new RegExp("Test", "g"), "\n")]];    
    await context.sync();  
  });
```
It replace "Test" with new line successful, but it lost my styles on text, its current result
![image](https://github.com/user-attachments/assets/13be40b5-1b41-4d37-ab9f-cc41031f179d)
What I need is below:
![image](https://github.com/user-attachments/assets/49bffdd3-35c5-4b59-b81e-ff4e4eb466b1)
Is there any way to set cell value with multiple lines without losing format

**Comment by adrianwu8516** [comment]
Hi @Edward-Zhou, as the rich text is not supported in Excel, there's no way to assign a rich text value to the cell to keep the format as what you demonstrated above. There is an already similar feature ask tracking this issue in the [tech community (3033874)](https://techcommunity.microsoft.com/t5/microsoft-365-developer-platform/richtext-support-excel-js-api/idi-p/3033874?search-action-id=714575805655&amp;search-result-uid=3033874) Please help vote on the post so that more people who also need this feature can notice, join the discussion, and vote for this. Thank you very much!!

**Comment by adrianwu8516** [comment]
Since the issues are already tracked on the above link, I will close this issue on GitHub.

**Comment by adrianwu8516** [comment]
Hi @Edward-Zhou, 

Thank you for your patience. The requested rich text support feature has been implemented and will be included in the upcoming Excel requirement set 1.18 release, scheduled for February 2025. In the meantime, you can already experiment with these capabilities in the beta build.

The following APIs will be available:

- `Excel.CellPropertiesInternal.TextRuns`
- `Excel.CellPropertiesLoadOptionsInternal.TextRuns`
- `Excel.RangeTextRun.Text`
- `Excel.RangeTextRun.Font`

For more details, please refer to the [official documentation](https://learn.microsoft.com/en-us/javascript/api/excel/excel.rangetextrun?view=excel-js-preview). Below is a sample code snippet that demonstrates how you can use rich text formatting in cells:

```javascript
(async () => {
  await Excel.run(async (context) => {
    var sheet = context.workbook.worksheets.getItem("TestSheet");
    const cellSrc = sheet.getRange("A1");
    const cellSrcTextRun: Excel.SettableCellProperties = {
      textRuns: [
        { text: "Sample", font: { bold: true, color: "#00B0F0", size: 14, italic: true } },
        { text: "1", font: { subscript: true } },
        { text: "String" },
        { text: "2", font: { superscript: true, color: "black", tintAndShade: 0.5 } },
      ],
    };

    cellSrc.clear(Excel.ClearApplyTo.all);
    cellSrc.setCellProperties([[cellSrcTextRun]]);
    await context.sync();

    const textRunCellProperty = cellSrc.getCellProperties({ textRuns: true });
    await context.sync();

    const cellTextRuns = textRunCellProperty.value[0][0].textRuns;
    console.log(JSON.stringify(cellTextRuns, undefined, "  "));
  });
})();
```

If you have any further questions or feedback, feel free to let us know!
